src/libsyntax_ext/proc_macro_registrar.rs

```diff
@@ -385,9 +389,21 @@ fn mk_registrar(cx: &mut ExtCtxt,
     let register_custom_derive = Ident::from_str("register_custom_derive");
     let register_attr_proc_macro = Ident::from_str("register_attr_proc_macro");
     let register_bang_proc_macro = Ident::from_str("register_bang_proc_macro");
+    let crate_kw = Ident::with_empty_ctxt(keywords::Crate.name());
+    let use_crate_kw = match cx.ecfg.features {
```

Comment:
Can't we always generate `crate::proc_macro_fn` regardless of editions and enabled features?
The whole registrar code is under `allow_internal_unstable: true` span anyway.

Reply:
I thought so! I originally did that but it generated feature errors :(

Reply:
That's because the path spans are set to `cd.span`/`ca.span`/`cm.span` and not `span`.
You can use `cd.span.with_ctxt(span.ctxt())` to keep the sane locations, but apply stability hygiene.

Reply:
Worked like a charm!